pyrometheus/chem_expr.py

```diff
@@ -229,9 +229,23 @@ def third_body_efficiencies_expr(sol: ct.Solution, react: ct.Reaction, c):
         of the species concentrations *c* as a
         :class:`pymbolic.primitives.Expression`
     """
-    efficiencies = [react.efficiencies[sp] for sp in react.efficiencies]
-    indices_nondef = [sol.species_index(sp) for sp in react.efficiencies]
-    indices_default = [i for i in range(sol.n_species) if i not in indices_nondef]
+    if isinstance(react, ct.ThreeBodyReaction):
+        from warnings import warn
+        warn("The 'Reaction.efficiencies' interface is deprecated and "
+             "will be removed after Cantera 3. Access efficiencies via "
+             "'react.third_body.efficiencies' instead")
```

Comment:
I'm not sure we need to warn here. There's no thing actionable for the pyro user here.

Reply:
But we do have some users that modify the codegen. Indeed, @jonathan-macart has one such PR #73 !